File: sagemaker/sagemaker-studio/IaC/case01/cdk-typescript/bin/logical-id-checker.ts
import * as cdk from 'aws-cdk-lib'
import { IConstruct } from 'constructs'

const regExp = /sagemaker/i

export class LogicalIdChecker implements cdk.IAspect {
  public visit(node: IConstruct): void {
    // Logical ID に "sagemaker" が含まれていないことをチェック
    if (node instanceof cdk.CfnResource) {
      if (regExp.test(node.logicalId)) {
        cdk.Annotations.of(node).addError('LogicalId contains "sagemaker"')
      }
    }
  }
}

File: sagemaker/sagemaker-studio/IaC/case01/cdk-python/chococones/logical_id_checker.py
import aws_cdk as cdk
from constructs import IConstruct
import jsii


@jsii.implements(cdk.IAspect)
class LogicalIdChecker:
    def visit(self, node):
        if isinstance(node, cdk.CfnResource):
            print("check")
            print(node.logical_id, type(node.logical_id))
            if "sagemaker" in node.logical_id.lower():
                cdk.Annotations.of(node).add_error('LogicalId contains "sagemaker"')
